### PR TITLE
Dump processes' output to standard output in addition to the log file

### PR DESCRIPTION
**Pull request by zakkak**
Pros:

1. Makes it easier to detect issues on CI, without having to download the build logs, extract, and look them up.
2. Makes it easier to see that there is progress and the build is not stuck.

Cons:

1. Makes the output a bit noisy.

### CODE REVIEW

File: testsuite/src/it/java/org/graalvm/tests/integration/utils/Commands.java [hunk 1]
```diff
@@ -216,6 +216,17 @@ public static void waitForExecutable(List<String> command, File directory) {
         }
     }
 
+    private static void dumpAndLogProcessOutput(File logFile, Process pA) throws IOException {
+        InputStream output = pA.getInputStream();
+        BufferedReader bufferedReader = new BufferedReader(new InputStreamReader(output));
```
Comment (jerboaa) [inline testsuite/src/it/java/org/graalvm/tests/integration/utils/Commands.java:221]: try-with-resources?

File: testsuite/src/it/java/org/graalvm/tests/integration/utils/Commands.java [hunk 2]
```diff
@@ -529,10 +540,11 @@ public void run() {
                 if (!log.exists()) {
                     Files.createFile(log.toPath());
                 }
-                Files.write(log.toPath(),
-                        ("Command: " + String.join(" ", command) + "\n").getBytes(), StandardOpenOption.APPEND);
-                pb.redirectOutput(ProcessBuilder.Redirect.appendTo(log));
+                String command = "Command: " + String.join(" ", this.command) + "\n";
+                System.out.println(command);
+                Files.write(log.toPath(), command.getBytes(), StandardOpenOption.APPEND);
```
Comment (jerboaa) [inline testsuite/src/it/java/org/graalvm/tests/integration/utils/Commands.java:545]: `getBytes(StandardCharsets.UTF-8)`?

**Review by jerboaa — APPROVED** [review]
Looks fine a few nits.

File: testsuite/src/it/java/org/graalvm/tests/integration/utils/Commands.java (hunk 1 above)
Comment (zakkak) [reply testsuite/src/it/java/org/graalvm/tests/integration/utils/Commands.java]: Good point, thanks!

**Comment by zakkak** [comment]
Converting to draft. It looks like this PR is causing CI to run indefinitely.
:eyes:

**Comment by zakkak** [comment]
The issue causing the stall has been fixed:

The issue was that the new dumpAndLogProcess is blocking, so it can't be used to start a web-server in the background and then start a client to query it. To avoid complicating things, I just don't use dumpAndLogProcess in `runCommand`, in those cases we just redirecto the output like we did before.

Another issue waiting to pop up was that dumpAndLogProcess didn't have a timeout, so if the underlying process was stuck the whole test would get stuck. This is now fixed by starting a single thread executor service for the dumping and logging with a timeout set.

@Karm this is now ready for you to re-review :)

**Comment by zakkak** [comment]
ping @Karm

**Comment by Karm** [comment]
THX @zakkak , I will take a look. I haven't acked it before as I was afraid of what the change does on Jenkins.  Gimme a day.

**Review by Karm — APPROVED** [review]
@zakkak Thank you :pray:    It didn't break anything on Jenkins and it seems logged in chronological order it seems.  Merging.

**Pull request merged**